### PR TITLE
Refactoring charge plug signals

### LINKED ISSUE #376: ChargingPort / ChargePlugType signals
I noticed that we have two signals related to charging port/plug. They should preferably be merged to a single signal, I see no value in keeping both. But what types (enum-literals) do we actually want to have? It seems that IEC 62106 standardizes plugs, but are literals standardized somewhere?

If one compare with Android it seems that we miss IEC Type 3 (deprecated so maybe does not matter that much) and some Tesla variants. Any reason not to have the same (number of) enums as Android?

```
Vehicle.Body.ChargingPort.Type:
  datatype: string
  type: attribute
  enum: [ "unknown", "Not_Fitted", "AC_Type_1", "AC_Type_2", "AC_GBT", "AC_DC_Type_1_Combo", "AC_DC_Type_2_Combo", "DC_GBT", "DC_Chademo" ]
  default: "unknown"
  description: Indicates the primary charging type fitted to the vehicle

Vehicle.Powertrain.TractionBattery.Charging.ChargePlugType:
  datatype: string
  type: attribute
  default: ccs
  enum: [ 'type 1', 'type 2', 'ccs', 'chademo' ]
  description: Type of charge plug available on the vehicle (CSS includes Type2).
```

Android definitions:
(from https://android.googlesource.com/platform/hardware/interfaces/+/master/automotive/vehicle/2.0/types.hal)
```
enum EvConnectorType : int32_t {
    /**
     * Default type if the vehicle does not know or report the EV connector
     * type.
     */
    UNKNOWN = 0,
    IEC_TYPE_1_AC = 1,              // aka Yazaki
    IEC_TYPE_2_AC = 2,              // aka Mennekes
    IEC_TYPE_3_AC = 3,              // aka Scame
    IEC_TYPE_4_DC = 4,              // aka CHAdeMO
    IEC_TYPE_1_CCS_DC = 5,          // aka Combo 1
    IEC_TYPE_2_CCS_DC = 6,          // aka Combo 2
    TESLA_ROADSTER = 7,
    TESLA_HPWC = 8,
    TESLA_SUPERCHARGER = 9,
    GBT_AC = 10,
    GBT_DC = 11,
    /**
     * Connector type to use when no other types apply. Before using this
     * value, work with Google to see if the EvConnectorType enum can be
     * extended with an appropriate value.
     */
    OTHER = 101,
};
```

### PR DESCRIPTION
**Pull request by erikbosch**
Fixes #376

Merging the two existing signals to a common signal
Inspired by EvConnectorType in
https://android.googlesource.com/platform/hardware/interfaces/+/master/automotive/vehicle/2.0/types.hal
See https://www.evexpert.eu/eshop1/knowledge-center/connector-types-for-ev-charging-around-the-world
for general information on connector types

Based on previous discussions in VSS I do not include any "unknown" literal in allowed values.

### CODE REVIEW

File: spec/Powertrain/Battery.vspec
Comment (SebastianSchildt) [inline spec/Powertrain/Battery.vspec:137]: allowed?

Comment (erikbosch) [reply spec/Powertrain/Battery.vspec:137]: Good catch :-)

**Review by gunnarx — CHANGES_REQUESTED** [review]

File: spec/Powertrain/Battery.vspec
Comment (erikbosch) [reply spec/Powertrain/Battery.vspec:164]: Now referencing to the applicable standards in the description. For now also including commonly used aliases for the connectors. But I have no problem in removing them if we prefer that. I added a reference to Wikipedia - there you can find most of the aliases listed.

Comment (SebastianSchildt) [reply spec/Powertrain/Battery.vspec:164]: No strong opinion, but I would switch the "length" of description and content, i.e.  short description to

```
description: IEC types refer to IEC 62196,  GBT refers to  GB/T 20234.
```
and put the rest in comment.

In the end, the signal itself is most important , and I have been and are fine with it "no matter" how we split the description exactly

**Review by SebastianSchildt — APPROVED** [review]

**Comment by erikbosch** [comment]
Meeting notes:

- Check Yaml specification on multiline comments/description before merging. Compare with https://yaml-multiline.info/

**Review by gunnarx — APPROVED** [review]

File: spec/Powertrain/Battery.vspec
Comment (gunnarx) [inline spec/Powertrain/Battery.vspec:154]: Should this multiline not use the Literal format, or is also this way OK according to spec?
Ref literals:  https://yaml.org/spec/1.2-old/spec.html#id2795688

**Review by gunnarx — CHANGES_REQUESTED** [review]

File: spec/Powertrain/Battery.vspec
Comment (erikbosch) [reply spec/Powertrain/Battery.vspec:154]: As I understand this format is "Yaml Plain style" see example 7.12 in  https://yaml.org/spec/1.2.2/#733-plain-style.
This is also described as "plain" in https://yaml-multiline.info/

So my interpretation is that this is valid Yaml.

With this format the output does not contain the newline used e.g.between line 151 and 152 above, i.e. it get merged to a single line. If using literal format (`|`) line breaks will be kept. We seem to support that correctly for vspec2yaml but not for vspec2csv as we then generate the linebreak rather than some form of escape (`\n`). But that is rather a bug in csv. In general I think it would be good if someone had time to check how the various tools handle various Yaml syntaxes, but that is not to be handled by this PR

**Comment by danielwilms** [comment]
meeting approved 15/02/22 after rebasing and single quotes.

**Comment by erikbosch** [comment]
@danielwilms - pr rebase/updated
@gunnarx - think you must approve or discard old review

File: spec/Powertrain/Battery.vspec
Comment (gunnarx) [reply spec/Powertrain/Battery.vspec:154]: Looks like it's valid YAML format then. OK, question answered.

**Review by gunnarx — APPROVED** [review]

**Pull request merged**
